extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/index.tsx

```diff
@@ -539,15 +371,56 @@ export const TransactionDetails = ({
     return isSwap ? t("Confirm Swap") : `${t("Confirm Send")}`;
   };
 
+  useEffect(() => {
+    const getData = async () => {
+      await fetchData();
+    };
+    getData();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const getBlockaidData = (
+    details: TxDetailsData | null,
+    source: ReturnType<typeof getAssetFromCanonical>,
+    dest: ReturnType<typeof getAssetFromCanonical>,
+  ) => {
+    const sourceBalance = findAssetBalance(details!.balances.balances, source);
+    const destBalance = findAssetBalance(
+      details!.destinationBalances.balances,
+      dest,
+    );
+    if (
+      sourceBalance &&
+      details!.isSourceAssetSuspicious &&
+      "blockaidData" in sourceBalance
+    ) {
+      return sourceBalance.blockaidData;
+    }
```

Comment:
if `sourceBalance` is suspicious, we're terminating early and we're not collecting any info about about `destBalance`. This subtly different than the logic it's replacing which would pass `isSuspicious === true` info about both assets if both were suspicious